File: backend/app/schemas/network.py
from pydantic import BaseModel
from typing import List

class Node(BaseModel):
    id: str
    name: str
    group: str  # 'person' or 'case'
    val: int    # Determines the size of the node visually
    category: str | None = None  # Added category
    date: str | None = None      # Added date
class Link(BaseModel):
    source: str
    target: str

class GraphData(BaseModel):
    nodes: List[Node]
    links: List[Link]

File: frontend/src/services/network.ts
import { api } from './api';

export interface Node {
  id: string;
  name: string;
  group: string;
  val: number;
  category?: string; // New field
  date?: string;     // New field
}

export interface Link {
  source: string;
  target: string;
}

export interface GraphData {
  nodes: Node[];
  links: Link[];
}

export const networkService = {
  getLinkAnalysis: async (): Promise<GraphData> => {
    const response = await api.get('/network/link-analysis');
    return response.data.data;
  }
};